File: csharp/GameOfLife.Engine/Position.cs
namespace GameOfLife.Engine
{
    public class Position
    {
        private Position(int x, int y)
        {
            Row = x;
            Column = y;
        }

        public int Row { get; }

        public int Column { get; }

        public static Position None => new Position(0, 0);

        public static Position From(int row, int column) => new Position(row, column);
    }
}

File: fsharp/GameOfLife.FSharp.Engine/Position.fs
namespace GameOfLife.FSharp.Engine
    
type Position = {
        row: int;
        column: int;
    } with

    static member none = {
        row = 0;
        column = 0;
    }

module Position =

    let create row column = {
        row = row;
        column = column;
    }

